Docs: Backend edits (#2059)
grafana/plugin-tools: docusaurus/docs/how-to-guides/data-source-plugins/profile-backend-plugin.md

  - back-end
---

> This guide provides instructions for configuring a plugin's backend to enable certain diagnostics when it starts, generating _profiling data_. Profiling data provides potentially useful information

Reviewer: ```suggestion
This guide provides instructions for configuring a plugin with a backend to enable certain diagnostics when it starts, generating _profiling data_. Profiling data provides potentially useful information
```

for investigating certain performance problems, such as high CPU or memory usage, or when you want to use [continuous profiling](https://grafana.com/oss/pyroscope/).

## Configure profiling data

The [Grafana configuration file](https://grafana.com/docs/grafana/latest/setup-grafana/configure-grafana/) allows you to configure profiling under the `[plugin.<plugin ID>]`.

In this section of the file, specify: 

> - The `<plugin ID>`, a unique identifier, for the backend you want to profile. For example, [grafana-github-datasource](https://grafana.com/grafana/plugins/grafana-github-datasource/). 

Reviewer: I think we can omit here we are talking about plugins with a backend since we have specified that before.

```suggestion
- The `<plugin ID>`, a unique identifier, for the plugin you want to profile. For example, [grafana-github-datasource](https://grafana.com/grafana/plugins/grafana-github-datasource/). 
```

## A note about overhead

> Running a plugin backend with profiling enabled and without [block](#the-profiling_block_rate-option) and [mutex](#the-profiling_block_rate-option) profiles enabled should only add a fraction of overhead. These endpoints are therefore suitable for production or continuous profiling scenarios.

Reviewer: ```suggestion
Running a plugin with profiling enabled and without [block](#the-profiling_block_rate-option) and [mutex](#the-profiling_block_rate-option) profiles enabled should only add a fraction of overhead. These endpoints are therefore suitable for production or continuous profiling scenarios.
```
